### ISSUE TITLE
ClayCard.Description requires a displayType prop but it "overrides" the tag prop

### ISSUE THREAD

**Issue by kresimir-coko** [liferay/clay#4154]
`ClayCard.Description` requires a `displayType` prop to be defined, but I have to specify `text` as the value, which is semantically wrong in order to get the size of the title to be how I want it to be. 

```js
<ClayCard.Description displayType="title" tag="h1">
	EVP Portlet
</ClayCard.Description>
```

Results in the element used for the title to be an `h1` but with the `card-title` class, which overrides the h1 size.
Here's how it looks:
![image](https://user-images.githubusercontent.com/24564752/123608519-1aba9580-d7ff-11eb-8404-6fcf3662da0b.png)

------

I'd like to have that same code produce the result that this code does
```js
<ClayCard.Description displayType="text" tag="h1">
	EVP Portlet
</ClayCard.Description>
```
which looks like:
![image](https://user-images.githubusercontent.com/24564752/123609281-d24fa780-d7ff-11eb-8301-d18835538fe2.png)

------

Bottom line is that if I want to have smaller, regular text intuitively I'd use `displayType="text"`, and when I want larger, heading sized text I'd use `displayType="title"`.

**Comment by matuzalemsteles** [comment]
Hey @kresimir-coko I investigated this quickly and maybe the problem to support this is that the `displayType` with `text` and `title` they force the font size which conflicts with setting the tag to `tag="h1"`, maybe instead to support this as part of the `Description` the best case here would be to just provide a `Heading` component, we are looking at this possibility in #4131.

Something like:

```jsx
<Heading level="2">EVP Portlet</Heading>
```

From my point of view, it doesn't seem very interesting to support this for `Description` because the class for it only forces fixed sizes, otherwise we would be making the `Description` just a `Heading` and also maybe the name is not semantic to what it does, maybe we were supposed to have a `Card.Title`, `Card.Text`...

**Comment by kresimir-coko** [comment]
Yea, it's up to you boys how you wanna do it. `ClayCard.Heading` was my initial thought when I was building the markup.

**Comment by matuzalemsteles** [comment]
Thank you, so to go with that now I think I would recommend not using `ClayCard.Description` with Heading tags due to CSS, in those cases it would be better to use the `h1` markup directly, we would have to do some breaking changes to support that or have to create new things and then we can leave that as a reinforcing issue for the `Heading` component implementation.

**Comment by matuzalemsteles** [comment]
Hey @kresimir-coko I'll close this issue in favor of #4659, now we have an idea of how to better handle typography in Clay, and let's create the component to handle it.
